src/libstd/io/mod.rs

```diff
@@ -1965,7 +1954,15 @@ impl<R: Read> Iterator for Bytes<R> {
     type Item = Result<u8>;
 
     fn next(&mut self) -> Option<Result<u8>> {
-        read_one_byte(&mut self.inner)
+        let mut byte = 0;
+        loop {
+            return match self.inner.read(slice::from_mut(&mut byte)) {
+                Err(ref e) if e.kind() == ErrorKind::Interrupted => continue,
+                Ok(0) => None,
+                Ok(..) => Some(Ok(byte)),
+                Err(e) => Some(Err(e)),
```

Comment:
Why the rearranging of the variants?

Reply:
Force of habit; I like putting my "early escapes" at the top. I can change it back to how it was.